### DISCUSSION TITLE
Tracking down instances of `LaTeX Error: Not in outer par mode.`

### DISCUSSION THREAD

**Discussion by phinate** [quarto-dev/quarto-cli#3635] [Q&A]
Hi! I've been working on a large document for some time now, and I've had a couple of instances where I've made figures using executable python code that have lead to this error message upon re-compilation. The figures appear to render fine while updating the preview dynamically.

Apologies since this is a bad issue (I can't come up with a good MWE right now) -- I was more looking for if you have an inkling as to the kinds of things that may cause this from a Quarto perspective? It's not totally obvious to me how these errors can manifest.

I'm only using the `#| fig-cap` and `#| label` attributes in those problematic figures, and they're not in any way super different from others that I'm generating in the same document.

Thanks, any pointers are appreciated! And sorry for being a little vague!

**Comment by cscheid** [comment]
I'm surprised about the re-compilation issue - we shouldn't get anything different between different runs of quarto.


Concretely, you get `Not in outer par mode` errors by doing things like putting a `\begin{figure}` inside a `\caption{}`.  So  in quarto, you get `Not in outer par mode` errors when you have "nested figures"; two constructs that are getting interpreted by quarto as figures.

If you want to track these more precisely, you can add the `keep-tex: true` option to your front matter, and inspect the resulting `.tex` file.
